app/reducers/article.js

```diff
@@ -12,26 +12,19 @@ function getInitialState() {
   return {
     article: {},
     currentArticle: null,
-    nextArticle: null,
-    topics: topicJsonMock.results
+    nextArticle: null
```

Comment:
Not sure what you are defining as an anti-pattern.  Which route are you talking about, the client route or the API route? Are you saying you don't want the saveArticleHighlight endpoint to suggest a next article to work on? Yeah, I should probably file that under "premature optimization is the root of all evil."

Let me expand on what I'm suggesting for API flow.

The server will be responsible for choosing which Annotator will get which article to work on, and the server cannot make that decision until it receives a request from the client for an article to work on. The client obviously will not have any idea what article ID will be returned.

Either at client startup or after saving a highlight, the client will ask for an article to work on from some sort of work queue endpoint, and the server will return either a full URL, or an URL hash # fragment, or an article ID. The front-end router will have to navigate appropriately to that URL so that it shows, just like it currently does.

When saving a highlight and requesting the next article, the client will have to wait on a promise from saveArticleHighlight to make sure that saveArticleHighlight doesn't return an error after the client navigates to the next article received from the endpoint "highlightQueue", or the person could start working on the next article, and the error on the previous article would be reported asynchronously, which would probably confused the UX. (What, my last article highlights didn't save, I'm being asked if I want to go back and fix???)

Does that clarify? Or are you thinking of a different API flow?